addons/account/tests/test_account_move_in_invoice.py

```diff
@@ -1088,6 +1088,116 @@ def test_in_invoice_line_onchange_currency_1(self):
             'amount_total': 208.01,
         })
 
+    def test_in_invoice_manual_currency_rate(self):
+        move_form = Form(self.invoice)
+        self.assertTrue(move_form._get_modifier('display_rate', 'invisible'))
+
+        move_form.currency_id = self.currency_data['currency']
+        move_form.save()
+
+        # Exchange rate is visible when using a foreign currency
+        self.assertFalse(move_form._get_modifier('display_rate', 'invisible'))
```

Comment:
```suggestion
        with Form(self.invoice) as move_form:
            self.assertTrue(move_form._get_modifier('display_rate', 'invisible'))

            move_form.currency_id = self.currency_data['currency']

            # Exchange rate is visible when using a foreign currency
            self.assertFalse(move_form._get_modifier('display_rate', 'invisible'))
```

Why do you need to save in between?